music-mapper/music-mapper
# new approach to oauth in boilermaker

Steps to use and test:
1. Go to localhost:8888 and click login with Spotify
2. We’re not correctly redirecting/reloading in react, but you will see the new uri of /auth/spotify/login (then just hit enter in the address bar to manually navigate to that uri)
3. /auth/spotify/login is a new express route that contains the Spotify authorization code that we were working with before. You can see the code in /server/auth/spotify.js and this is where we use the App’s client_id, redirect_uri, etc…
4. Then the user should get shown the Spotify login page, and once they submit they will be redirected to localhost:8888
5. When the user logs in, I added a global variable that keeps track of the Spotify access_token so that we can use it for other API requests. To see this working, manually navigate to /api/spotify/check which is another new API route that uses the access_token to request the User’s list of albums (this would maybe be what it would look like if we added an API route for get-data, and maybe there be a button for this on the homepage. i.e. see your data?)
6. When the api/spotify/check route finishes, it will redirect to homepage again
7. In your terminal you will see console log data
NOTE: I added the client secret, client id and redirect uri as environment variables rather than in a secrets file.
I also wanted to edit the Redirect URI so I created a new app on Spotify. If you want to test the code you could just update the old app’s uri (change both your environment variable and on Spotify’s end by logging in to the app online), or just switch everything over to the new app Client ID and Secret that I am using by overwriting your environment variables
    
export SPOTIFY_CLIENT_ID='119b8f5740114efb8ee2c2cafacd2bbd'
export SPOTIFY_CLIENT_SECRET='c21c21dac51f475c8f5a7bddf8469748'
export REDIRECT_URI='http://localhost:8888/auth/spotify/callback'
